### DISCUSSION TITLE
What should node_modules include after building for serverless

### DISCUSSION THREAD

**Discussion by imoby** [prisma/prisma#19386] [Q&A]
Hey all, I've posted a issue ticket around this and I wanted to also mention here because I wanted to better understand what the expectation should be. I have a serverless nodejs with postgres using prisma setup. I am using serverless-webpack-prisma to build this and each of my lambdas when deployed have a node_modules with the following below. I'm curious based on the comments I mentioned this issue ticket whether the output is correct especially because the size of my lambda is quite large and the commenter suggested that I do not include the prisma package, only the prisma client but I am not manually adding it anywhere so I'm not sure what I'm needing to do here: https://github.com/prisma/prisma/issues/19358

![image](https://github.com/prisma/prisma/assets/6380855/55d114a6-f040-490a-8662-6918cbced02a)
![image](https://github.com/prisma/prisma/assets/6380855/a10db74d-1a20-4547-ab8e-4f01715e057e)
![image](https://github.com/prisma/prisma/assets/6380855/bff7fc15-e617-4bec-9dbe-d6d52c528217)

**Comment by nurul3101** [comment]
Hello @imoby 👋 

I believe you are referring to this [comment](https://github.com/prisma/prisma/issues/19358#issuecomment-1556781901) that mentions removing prisma. Can you please share you `package.json`? Is prisma listed in the devDependencies?

**Comment by imoby** [comment]
Hey @nurul3101 as I mentioned in the issue itself, will post it here too:

I can definitely try to work on getting a reproducible example at some point but essentially this is the contents of my specific lambda bundles package.json file and I have also included the package.json file of my project itself which only has the prisma client not the prisma cli

```
{
  "name": "sample",
  "version": "1.0.0",
  "description": "Packaged externals for sample",
  "private": true,
  "scripts": {},
  "dependencies": {
    "@prisma/client": "^4.14.1",
    "zod": "^3.17.3"
  }
}
```
[node_modules.zip](https://github.com/prisma/prisma/files/11560920/node_modules.zip)

With that, I am seeing the attached node_modules (I've compressed and attached it to this). I'm curious why it is such a large package and why my lambda is seeing such a high package size when my actual lambda function code itself is just 6kb

<img width="345" alt="image" src="https://github.com/prisma/prisma/assets/6380855/9035f423-6a89-4541-a089-1c78308c12e8">

----------

Also below is my top level serverless package json file (minus some things commented out for privacy):

```
{
  "name": "sample",
  "version": "1.0.0",
  "scripts": {
  },
  "dependencies": {
    "@aws-sdk/client-dynamodb": "^3.266.1",
    "@aws-sdk/client-eventbridge": "^3.298.0",
    "@aws-sdk/util-dynamodb": "^3.266.1",
    "@prisma/client": "^4.14.1",
    "@serverless/typescript": "^3.19.0",
    "apollo-server-lambda": "^3.9.0",
    "dayjs": "^1.11.7",
    "dotenv-cli": "^5.1.0",
    "global": "^4.4.0",
    "graphql": "^16.5.0",
    "serverless-dynamodb-local": "^0.2.40",
    "serverless-http": "^3.0.1",
    "zod": "^3.17.3"
  },
  "devDependencies": {
    "@commitlint/cli": "^16.3.0",
    "@commitlint/config-conventional": "^16.2.4",
    "@types/aws-lambda": "^8.10.101",
    "@types/jest": "^29.4.0",
    "@types/node": "^18.0.1",
    "@types/serverless": "^3.12.9",
    "@types/supertest": "^2.0.12",
    "@typescript-eslint/eslint-plugin": "^5.27.0",
    "@typescript-eslint/parser": "^5.27.0",
    "cross-env": "^7.0.3",
    "eslint": "^8.16.0",
    "eslint-config-airbnb-base": "^15.0.0",
    "eslint-config-airbnb-typescript": "^17.0.0",
    "eslint-config-prettier": "^8.5.0",
    "eslint-plugin-import": "^2.26.0",
    "eslint-plugin-jest": "^26.4.6",
    "eslint-plugin-jest-formatting": "^3.1.0",
    "eslint-plugin-prettier": "^4.0.0",
    "eslint-plugin-simple-import-sort": "^7.0.0",
    "eslint-plugin-unused-imports": "^2.0.0",
    "husky": "^7.0.4",
    "jest": "^29.4.1",
    "lint-staged": "^12.5.0",
    "prettier": "^2.6.2",
    "rimraf": "^3.0.2",
    "serverless": "^3.31.0",
    "serverless-bundle": "^6.0.0",
    "serverless-dotenv-plugin": "^4.0.1",
    "serverless-offline": "^12.0.3",
    "serverless-webpack": "^5.11.0",
    "serverless-webpack-prisma": "^1.1.2",
    "supertest": "^6.2.3",
    "ts-jest": "^28.0.4",
    "ts-loader": "^9.4.2",
    "ts-node": "^10.8.2",
    "typescript": "^4.7.2",
    "webpack": "^5.75.0",
    "webpack-node-externals": "^3.0.0"
  }
}
```

**Reply by janpio** [reply]
No need to discuss the exact same thing in two locations.
Let's continue in https://github.com/prisma/prisma/issues/19358

**Reply by nurul3101** [reply]
I'll mark this discussion as closed. As Jan mentioned we can continue the conversation on #19358
